server/app.es6

```diff
@@ -1,17 +1,376 @@
+  };
+};
+
+let translateArrayFromDatabaseToJSON = (applications) => {
+  for (var i = 0; i < applications.length; i++) {
+    applications[i] = translateFromDatabaseToJSON(applications[i]);
+  }
+  return applications;
+};
+
+// populates an applicationId on the object before return
 let createNoncommercialTempApp = (req, res) => {
-  req.body['tempControlNumber'] = 1;
-  res.status(200).json(req.body);
-}
+
+  let errorArr = [];
+  let errorRet = {};
+
+  // overall validation
+  let result = v.validate(req.body, schemas.noncommercialSchema, validatorOptions);
+  if (result.errors.length > 0) {
+    collateErrors(result, errorArr);
+  }
+
+  // if there is an evening phone, validate it
+  if (req.body.applicantInfo.eveningPhone && Object.keys(req.body.applicantInfo.eveningPhone).length > 0) {
+    result = v.validate(req.body.applicantInfo.eveningPhone, schemas.phoneNumberSchema, validatorOptions);
+    collateErrors(result, errorArr, 'applicantInfo.eveningPhone.');
+  }
+
+  // if the orgType is Individual, then primaryAddress is required
+  if (req.body.applicantInfo.orgType === 'Individual') {
+    if (req.body.applicantInfo.primaryAddress && Object.keys(req.body.applicantInfo.primaryAddress).length > 0) {
+      result = v.validate(req.body.applicantInfo.primaryAddress, schemas.addressSchema, validatorOptions);
+      collateErrors(result, errorArr, 'applicantInfo.primaryAddress.');
+    } else {
+      errorArr.push('required-applicantInfo.primaryAddress');
+    }
+  }
+
+  // if the orgType is Corporation, then organizationAddress is required and might have a primary address
+  if (req.body.applicantInfo.orgType === 'Corporation') {
+    if (req.body.applicantInfo.organizationAddress && Object.keys(req.body.applicantInfo.organizationAddress).length > 0) {
+      result = v.validate(req.body.applicantInfo.organizationAddress, schemas.addressSchema, validatorOptions);
+      collateErrors(result, errorArr, 'applicantInfo.organizationAddress.');
+    } else {
+      errorArr.push('required-applicantInfo.organizationAddress');
+    }
+
+    if (req.body.applicantInfo.primaryAddress && Object.keys(req.body.applicantInfo.primaryAddress).length > 0) {
+      result = v.validate(req.body.applicantInfo.primaryAddress, schemas.addressSchema, validatorOptions);
+      collateErrors(result, errorArr, 'applicantInfo.primaryAddress.');
+    }
+  }
+
+  // if secondaryAddress exists, then validate it
+  if (req.body.applicantInfo.secondaryAddress && Object.keys(req.body.applicantInfo.secondaryAddress).length > 0) {
+    result = v.validate(req.body.applicantInfo.secondaryAddress, schemas.addressSchema, validatorOptions);
+    collateErrors(result, errorArr, 'applicantInfo.secondaryAddress.');
+  }
+
+  if (errorArr.length > 0) {
+    //console.log(errorArr);
+
+    errorRet['errors'] = errorArr;
+
+    res.status(400).json(errorRet);
+  } else {
+
+    // create the noncommercial app object and persist
+
+    NoncommercialApplication.create({
```

Comment:
Not sure what you mean by this.  We're creating the sequelize object using the sequelize model.